handlers/search.go

```diff
@@ -76,47 +76,43 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
 
 	switch searchType {
 	case "slots":
-		if len(search) <= 1 {
+		if len(search) <= 1 || !searchLikeRE.MatchString(search) {
 			break
 		}
 		result = &types.SearchAheadSlotsResult{}
-		if searchLikeRE.MatchString(search) {
-			if _, convertErr := strconv.ParseInt(search, 10, 32); convertErr == nil {
-				err = db.ReaderDb.Select(result, `
-				SELECT slot, ENCODE(blockroot, 'hex') AS blockroot 
-				FROM blocks 
-				WHERE slot = $1
-				ORDER BY slot LIMIT 10`, search)
-			} else if len(search) == 64 {
-				blockHash, err := hex.DecodeString(search)
-				if err != nil {
-					logger.Errorf("error parsing blockHash to int: %v", err)
-					http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-					return
-				}
-				err = db.ReaderDb.Select(result, `
-				SELECT slot, ENCODE(blockroot, 'hex') AS blockroot 
-				FROM blocks 
-				WHERE blockroot = $1 OR
-					stateroot = $1
-				ORDER BY slot LIMIT 10`, blockHash)
-				if err != nil {
-					logger.Errorf("error reading block root: %v", err)
-					http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-					return
-				}
+		if _, convertErr := strconv.ParseInt(search, 10, 32); convertErr == nil {
+			err = db.ReaderDb.Select(result, `
+			SELECT slot, ENCODE(blockroot, 'hex') AS blockroot 
+			FROM blocks 
+			WHERE slot = $1
+			ORDER BY slot LIMIT 10`, search)
+		} else if len(search) == 64 {
+			var blockHash []byte
+			blockHash, err = hex.DecodeString(search)
+			if err != nil {
+				err = fmt.Errorf("error parsing blockHash to int: %v", err)
+				break
 			}
+			err = db.ReaderDb.Select(result, `
+			SELECT slot, ENCODE(blockroot, 'hex') AS blockroot 
+			FROM blocks 
+			WHERE blockroot = $1 OR
+				stateroot = $1
+			ORDER BY slot LIMIT 10`, blockHash)
 		}
 	case "blocks":
-		number, err := strconv.ParseUint(search, 10, 64)
+		var number uint64
+		number, err = strconv.ParseUint(search, 10, 64)
+		if err != nil {
+			break
+		}
+		var block *types.Eth1Block
+		block, err = db.BigtableClient.GetBlockFromBlocksTable(number)
```

Comment:
This now adds log entries like `level=error msg="error doing query for searchAhead" error="block not found" module=handlers searchType=blocks` (searched for 9999999999).

Reply:
should be fixed, ptal